r/tests/testthat/test-dplyr-join.R

```diff
@@ -90,8 +90,56 @@ test_that("Error handling", {
     left_tab %>%
       left_join(to_join, by = "not_a_col") %>%
       collect(),
-    "all(names(by) %in% names(x)) is not TRUE",
-    fixed = TRUE
+    "Join columns must be present in data"
+  )
+
+  # we print both message_x and message_y with an unnamed `by` vector
```

Comment:
I like the comments here that explain _why_ these tests are here / what is being checked — though part of me wonders if this is showing the downside to snapshots (not having the error here in the code showing what we are looking). 

Let's keep the snapshots for now (especially cause greping the `rlang::abort()` messages is not super fun). Is there a Jira already for adding something to our devdocs about how to use snapshots/our philosophy on how to review them/limitations (e.g. C++ tests not working well)/that kind of stuff? We shouldn't expand the scope of this PR for just that, but it would be good to make a jira that adds that info so we all agree how we will use them going forward.

Reply:
I think a [relevant Jira](https://issues.apache.org/jira/browse/ARROW-14458) already exists. We might not need another one, but rather update it to reflect current thinking (i.e. use `expect_snapshot(error = TRUE)` where possible, the future addition of classed errors to cpp11, etc). What do you think?

Reply:
That Jira is great, please to update it to reflect this thinking